modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsController.scala

```diff
@@ -290,4 +291,54 @@ object TcsController {
     implicit val equal: Eq[Subsystem] = Eq.fromUniversalEquals
   }
 
+  sealed trait GuidersConfig[+C] {
+    val pwfs1: GuiderConfig@@P1Config
+    val pwfs2: GuiderConfig@@P2Config
+    val oiwfs: GuiderConfig@@OIConfig
+  }
+
+  @Lenses
+  final case class BasicGuidersConfig(
+    pwfs1: GuiderConfig@@P1Config,
+    pwfs2: GuiderConfig@@P2Config,
+    oiwfs: GuiderConfig@@OIConfig
+  ) extends GuidersConfig[Nothing]
+
+  @Lenses
+  final case class AoGuidersConfig[C](
+    pwfs1: GuiderConfig@@P1Config,
+    aoguide: C,
+    oiwfs: GuiderConfig@@OIConfig
+  ) extends GuidersConfig[C] {
+    override val pwfs2: GuiderConfig@@P2Config =
+      tag[P2Config](GuiderConfig(ProbeTrackingConfig.Parked, GuiderSensorOff))
+  }
+
+  sealed trait TcsConfig[+C, +G] {
```

Comment:
Why this one is covariant on C and G?

Reply:
First, I need two types because `AoTcsConfig` needs two types ( I plan to remove one ), and if I don't put both in `TcsConfig`, then I will not be able to match. And they are both covariant, so I can define BasicTcsConfig as a subtype of every TcsConfig.